### PR TITLE
[KOGITO-478] Add possibility to run e2e tests with external image 

### PR DESCRIPTION
**Pull request by sutaakar**
Set operator image tag based on tag parameter

https://issues.jboss.org/browse/KOGITO-478

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by sutaakar** [comment]
Still thinking about this a bit.
What should be the usage of these e2e tests?
Should they support execution of local implementation (tests would have to run sequentially, tests will have to take care of cleaning up the namespace in the end) or rather focus more or executing tests against specific existing images (allowing parallel test runs in separate namespaces)?
Or both?

**Comment by ricardozanini** [comment]
@sutaakar thanks for your time having a look at this. Let me try to reply your questions.

> What should be the usage of these e2e tests?
For now, I'm using it as a final assessment before submiting a PR. In the future I intend to have those to be run in CI, so we could automate at least basic usage if this tests.

> Should they support execution of local implementation (tests would have to run sequentially, tests will have to take care of cleaning up the namespace in the end) or rather focus more or executing tests against specific existing images (allowing parallel test runs in separate namespaces)?
Or both?
I'd say that we can have those tests to check basic use cases in complex environments. This means deploying a native and non-native Kogito service on OCP 3.11, 4.x and Kubernetes, for example. With persitence and without. With data index and without and so on.

File: hack/run-e2e.sh [hunk 1]
```diff
@@ -27,19 +27,17 @@ fi
...
   echo "-------- tag is empty, assuming default from the Operator"
+else
+  E2E_PARAMS="${E2E_PARAMS} --image quay.io/kiegroup/kogito-cloud-operator:${tag}"
 fi
 
 echo "-------- Running e2e tests with namespace=${namespace}, tag=${tag}, native=${native} and maven_mirror=${maven_mirror}"
 
 # creates the namespace
 oc create namespace ${namespace}
-# gives permissions
```
Comment (ricardozanini) [inline hack/run-e2e.sh]: Why did you remove this? Without the required permissions the operator won't run correctly.

Comment (sutaakar) [reply hack/run-e2e.sh]: Originally I thought that these permissions are handled by operator-sdk when running the tests. But as I check the documentation, the role.yaml and role_binding.yaml are not applied. I will return it back.

**Comment by sutaakar** [comment]
Adjusted implementation.
Now it is possible to run tests against specific docker image or using local operator code.
In case of docker image the service accounts and roles are created by operator-sdk.
Local operator code functionality stays as it is.

**Comment by ricardozanini** [comment]
@sutaakar could you please open a JIRA to track this change?

**Comment by sutaakar** [comment]
Adjusted commit message with JIRA reference
